### ISSUE TITLE
New rule: eslint-disable-next-line ...

### ISSUE THREAD

**Issue by bionikspoon** [eslint/eslint#5206]
**Premise**

With the `// eslint-disable-line ...` comment, lines can get  a bit long.  It'd be nice to be able to put the comment on the preceding line.

**When does this rule warn? Please describe and show example code:**

For example:

```js
// eslint-disable-line no-extend-native
Map.prototype.updateGrid = function updateGrid(col, row) { 
  return this.merge({ col, row });
};
```

Instead of:

```js
Map.prototype.updateGrid = function updateGrid(col, row) { // eslint-disable-line no-extend-native
  return this.merge({ col, row });
};
```

It'd be great if the parser was smart enough to recognize when the line is effectively blank, it's referring to the next line.  However, in case that's unreasonable, let me propose a new rule: `// eslint-disable-next-line ...`

For example:
```js
// eslint-disable-next-line no-extend-native
Map.prototype.updateGrid = function updateGrid(col, row) { 
  return this.merge({ col, row });
};
```

**Is this rule preventing an error or is it stylistic?**

Stylistic.  

**Why is this rule a candidate for inclusion instead of creating a custom rule?**

Not sure if it can be written as a stand-alone rule.

**Are you willing to create the rule yourself?**

No.  I'm not sure this is in my skill set.

**Comment by eslintbot** [comment]
@bionikspoon Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
We already support ```/*eslint-disable */``` ```/*eslint-enable */``` comments. As well as ```/*eslint*/``` comments. Are those not sufficient for your usecase? If not, can you explain why? Also, keep in mind this isn't a rule, this would have to be done in the core.

**Comment by BYK** [comment]
@ilyavolodin I think the reason is `/*eslint-disable */` requires a matching `/*eslint-enable */` afterwards and doing that only for a single line, just because the line gets too long with the `// eslint-disable-line ` directive is a bit wasteful. At least that's how I feel and why I'd support something like this.

@bionikspoon is that accurate?

**Comment by bionikspoon** [comment]
@BYK @ilyavolodin Yes!  The idea is to reduce visual noise from your code.  Instructions for your build tools should be relegated to the background.

Long lines can be noisy, and especially noisy when the content is ancillary.  I tried to illustrate this.    If you use text wrapping in your editor, it's even noisier as it effectively takes another line.  As @BYK says, wrapping your code in an open-close block for a one-off rule-break is wasteful and, I would add, noisy.

My suggestion is purely an appeal to visual aesthetics, which I think is in bounds for a lint tool.  We all want beautiful code :smile: .

**Comment by platinumazure** [comment]
For what it's worth, I would support this too.

**Comment by ilyavolodin** [comment]
We already have 3 different ways to turn off rules in the comments. I'm not sure we want fourth way. @eslint/eslint-team what do you think?

**Comment by jaydeep17** [comment]
+1

**Comment by nzakas** [comment]
I think it's fine, but we will need someone (outside the ESLint team) to contribute this change.

**Comment by kaicataldo** [comment]
I'll try to take a look at this over the next week

**Comment by kaicataldo** [comment]
Actually, was pretty simple to figure out! PR opened.

**Comment by bionikspoon** [comment]
@kaicataldo awesome!  Glad to hear this idea panned out.
